presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergSplitManager.java

```diff
@@ -69,7 +70,7 @@ public ConnectorSplitSource getSplits(
         }
 
         Table icebergTable;
-        if (nativeCatalogMode) {
+        if (catalogType == HADOOP) {
             icebergTable = resourceFactory.getCatalog(session).loadTable(toIcebergTableIdentifier(table.getSchemaTableName()));
         }
         else {
```

Comment:
might be out of scope of this PR, I'm thinking make this getIcebergTable logic to be a utility method or in polymorphism

Reply:
Agree! This will also help to better manage the creation of the two metadata instances. As this PR is just about merging the native-mode config into the catalog type, I'll send follow-up PRs for this new abstraction.

Reply:
Agree! I created a utility function to get the iceberg table. 
Actually there are some other places such as `IcebergMetadata` and `IcebergNativeMetadata` where we could do some optimization by creating new abstraction layers. As this PR just focuses on merging the native-mode config into the catalog type, I'll send some follow-up PRs for these new abstraction layers.